### ISSUE TITLE
GraphQL mutations to setup notifications for entities

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#1128]
For each entity (project, branch, ...), provide a mutation to setup its notifications. For example:

```yaml
setBranchNotifications(input: {
  project: "my-project",
  branch: "my-branch",
  authoritative: true,
  notifications: [
    {
      channel: "jenkins",
      config: {
         // ...
      }
    }
  ]
})
```

Variants per ID (for example `setBranchNotificationsById(input: {branchId...})` are also to be provided.

The `authoritative` flag is there to tell if the list of provided notifications is the _unique_ list of notifications for this entity, or if this list is only appended (duplicates will be replaced).

Additionally, adapt the Ontrack Jenkins pipeline library to use these mutations at project & branch level to allow the setup of notifications at project or branch level in the `ontrackCliSetup` task.

Provide also new commands (for project & branch only) in the Ontrack CLI.

Note that the GitHub ingestion already allows for the setup of notifications at entity level.

**Comment by dcoraboeuf** [comment]
There is already a `subscribeToEvents` mutation, which is already idempotent.

What is missing is the possibility to target a project entity by name, like `subscribeBranchToEvents`.

**Comment by dcoraboeuf** [comment]
Available in 4.7.23
